### PR TITLE
Added config items for late block reorg

### PR DESCRIPTION
**Pull request by rolfyone**
- added  `REORG_HEAD_WEIGHT_THRESHOLD`, `REORG_PARENT_WEIGHT_THRESHOLD`, and  `REORG_MAX_EPOCHS_SINCE_FINALIZATION` to phase0 configuration, and defaulted

 - added utility functions:
   - `isShufflingStable`
   - `isFinalizationOk`
   - `calculateCommitteeFraction`

which were part of the changes in https://github.com/ethereum/consensus-specs/pull/3034

partially addresses #6595

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/config/SpecConfigPhase0.java [hunk 1]
```diff
@@ -116,6 +116,10 @@ public class SpecConfigPhase0 implements SpecConfig {
   private final int attestationSubnetCount;
   private final int attestationSubnetExtraBits;
   private final int attestationSubnetPrefixBits;
+  private final int reorgMaxEpochsSinceFinalization;
+
```
Comment (lucassaldanha) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/config/SpecConfigPhase0.java]: Any reason for this empty line?

File: ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/util/ForkChoiceUtilTest.java [hunk 2]
```diff
@@ -307,6 +312,52 @@ void canOptimisticallyImport_shouldBeFalseWhenBlockIsMergeButNotOldEnough() {
...
+  public static Stream<Arguments> isShufflingStableConditions() {
+    // slot , expectedResult
+    ArrayList<Arguments> args = new ArrayList<>();
+    for (int i = 80; i < 100; i++) {
+      // every start of epoch, shuffling is considered not stable
+      args.add(Arguments.of(i, i % 8 != 0));
+    }
+
+    return args.stream();
+  }
```
Comment (lucassaldanha) [inline ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/util/ForkChoiceUtilTest.java]: I understand what you did here, but from a readability point of view it isn't as clear as if you had a few entries of different scenarios (similar to what you did on `isFinalizationOkConditions`).

**Review by lucassaldanha — APPROVED** [review]
LGTM. Just some readability comments.

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/config/SpecConfigPhase0.java (hunk 1 above)
Comment (rolfyone) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/config/SpecConfigPhase0.java]: hrm not really, i can clean that up, i just did the first one then the second two

File: ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/util/ForkChoiceUtilTest.java (hunk 2 above)
Comment (rolfyone) [reply ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/util/ForkChoiceUtilTest.java]: its just a lot of copy/paste. i might keep the loop for the false.. will see if i can make it clear still

**Pull request merged**